lib/db/mem.js

```diff
@@ -224,6 +254,19 @@ module.exports = function (log, error) {
     return P.resolve({})
   }
 
+  Memory.prototype.deleteDevice = function (uid, deviceId) {
+    return getAccountByUid(uid)
+      .then(
+        function (account) {
+          var id = internalDeviceId(account, { id: deviceId })
+          var device = devices[id] || {}
+          delete sessionTokens[device.sessionTokenId]
+          delete account.devices[device.id]
+          return {}
```

Comment:
Should this also delete it from the global `devices` object?

Actually, it's not clear to me what that global `devices` object is for, since you should always have a reference to the account when dealing with a device.